shoop_tests/core/test_fields.py

```diff
@@ -7,10 +7,38 @@
 # LICENSE file in the root directory of this source tree.
 import pytest
 from django.core.exceptions import ImproperlyConfigured
+from django.forms import ModelForm
 
 from shoop.core.fields import MeasurementField
+from shoop.core.models import Product
+from shoop.testing.factories import get_default_product, get_default_shop
 
 
 def test_measurement_field_doesnt_know_bananas():
     with pytest.raises(ImproperlyConfigured):
         scale = MeasurementField(unit="banana")
+
+@pytest.mark.django_db
+def test_formatted_decimal_field():
+    """
+    Test that FormattedDecimalField doesn't return value in scientific
+    notation.
+    """
+
+    class TestModelForm(ModelForm):
+        class Meta:
+            model = Product
+            fields = ["width"]
+
+    get_default_shop()
+
+    product = get_default_product()
+    product.width = 0
+    product.save()
+    product = Product.objects.first()
+
+    form = TestModelForm(instance=product)
+    rendered_form = str(form)
+
+    assert "value=\"0E-9\"" not in rendered_form
+    assert "value=\"0\"" in rendered_form
```

Comment:
This test should be done without using the db, since it doesn't really need db for anything: just don't save the product and initialize its width to `Decimal('0E-9')`.

You should also parameterize this test for other values than just '0E-9' too, IMHO at least following should be tested `'0E-9', '0E-100', '1E-9', '-1E-9'` and also some values with many different digits in their value. Here's a list that I just tried in console:
```python
values_str = '0E-9 0E-30 1E-9 123E-10 -123E-10 1.12345666666666E20 1.23E-10000'
[(str(val), format(val, 'f')) for val in [decimal.Decimal(x) for x in values_str.split()]]
```

The last one is nasty, since its string representation contains 10004 characters. Also checked that it's possible to have even decimal values that generate 100 megabytes of characters when formatted: `len(format(decimal.Decimal('1.23E-100000000'), 'f'))`.  Probably the field should have some kind of maximum length. Maybe it could be truncated to something like 100 characters (`format(val, 'f')[:100]`). What do you think?